README.md

```diff
@@ -1,5 +1,131 @@
-# Minishift Fedora VM
+<a name="creating-a-minishift-fedora-iso"></a>
+# Minishift Fedora ISO
 
 This repository contains all the instructions and code to build a Live ISO based on Fedora
 which can be used by [minishift](https://github.com/minishift/minishift) as an alternative to
-the boot2docker and CentOS ISO.
+the boot2docker ISO.
+
+----
+
+<!-- MarkdownTOC -->
+
+- [Building the Fedora ISO](#building-the-fedora-iso)
+	- [On Fedora](#on-fedora)
+		- [Prerequisites](#prerequisites)
+		- [Building the ISO](#building-the-iso)
+	- [On hosts _other than Fedora_ \(OS X, Windows, CentOS ...\)](#non-fedora-hosts)
+		- [Prerequisites](#prerequisites-1)
+		- [Building the ISO](#building-the-iso-1)
+  - [Manual release](#manual-release)
+- [Further reading](#further-reading)
+- [Community](#community)
+
+<!-- /MarkdownTOC -->
+
+----
+
+<a name="building-the-fedora-iso"></a>
+## Building the Fedora ISO
+
+The following contains instructions on how to build the default (Fedora based) ISO.
+If you are able to install [livecd-tools](https://github.com/rhinstaller/livecd-tools)
+directly on your machine, you can use the [Fedora](#on-fedora) instructions.
+
+If you don't have _livecd-tools or using different linux distro other than fedora_, follow the
+[hosts other than Fedora](#non-fedora-hosts) instructions.
+
+<a name="on-fedora"></a>
+### On Fedora
+
+<a name="prerequisites"></a>
+#### Prerequisites
+* Update your system before start and if there is kernel update then reboot your system to activate latest kernel.
+
+        $ dnf update -y
+
+* [Install livecd-tools](https://github.com/rhinstaller/livecd-tools)
+
+  Note: We use to have docker installed on system to get selinux context, [check bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1303565)
+
+        $ dnf install -y livecd-tools docker
+
+
+<a name="building-the-iso"></a>
+#### Building the ISO
+
+```
+$ git clone https://github.com/minishift/minishift-fedora-iso.git
+$ cd minishift-fedora-iso
+$ make
+```
+
+<a name="non-fedora-hosts"></a>
+### On hosts _other than Fedora_ (OS X, Windows, CentOS ...)
```

Comment:
Let's remove `Windows`. We have never tested it as I remember.
Did you @praveenkumar @LalatenduMohanty @gbraad @agajdosi ?

If we can't sure, we shouldn't mention it.
Also, I am not interested in building ISO in Windows at all :smile:

Reply:
Change `OS X` to `macOS`.

Reply:
Vagrant and Windows works well... I would leave it in

Reply:
Just want to be consistent here. :) Its difficult to compile binary in Windows.

Reply:
We are NOT building a binary here, but the ISO. and so far we have described each ISO to be able to be build using Vagrant. Besides, we might have to do this anyway, as on CentOS you have to use a VM to build the Fedora ISO... you can not rely on CentOS to provide a workable result.

Reply:
Besides, customization of the ISO is a possiblity that people want, and this has nothing to do with the need to build the actual minishift binary.

Reply:
@gbraad , If it is working in Windows then I am fine :)